webknossos-datastore/app/com/scalableminds/webknossos/datastore/explore/ExploreLocalLayerService.scala

```diff
@@ -42,25 +49,67 @@ class ExploreLocalLayerService @Inject()(dataVaultService: DataVaultService)
+            makeZarrSegmentationLayerPathRelative(l)
+      },
+      new NgffExplorer
+    )(path, dataSourceId, "")
+
+  private def exploreLocalNeuroglancerPrecomputed(path: Path, dataSourceId: DataSourceId, layerDirectory: String)(
+      implicit ec: ExecutionContext): Fox[DataSource] =
+    exploreLocalLayer(
+      layers =>
+        layers.map {
+          case l: PrecomputedDataLayer => l.copy(mags = l.mags.map(m => m.copy(path = m.path.map(_.split("/").last))))
+          case l: PrecomputedSegmentationLayer =>
+            l.copy(mags = l.mags.map(m => m.copy(path = m.path.map(_.split("/").last))))
+      },
+      new PrecomputedExplorer
+    )(path, dataSourceId, layerDirectory)
+
+  private def exploreLocalN5Multiscales(path: Path, dataSourceId: DataSourceId, layerDirectory: String)(
+      implicit ec: ExecutionContext): Fox[DataSource] =
+    exploreLocalLayer(
+      layers =>
+        layers.map {
+          case l: N5DataLayer         => l.copy(mags = l.mags.map(m => m.copy(path = m.path.map(_.split("/").last))))
+          case l: N5SegmentationLayer => l.copy(mags = l.mags.map(m => m.copy(path = m.path.map(_.split("/").last))))
+      },
+      new N5MultiscalesExplorer
+    )(path, dataSourceId, layerDirectory)
+
+  private def exploreLocalLayer(makeLayersRelative: List[DataLayer] => List[DataLayer], explorer: RemoteLayerExplorer)(
+      path: Path,
+      dataSourceId: DataSourceId,
+      layerDirectory: String)(implicit ec: ExecutionContext): Fox[DataSource] =
     for {
-      remoteSourceDescriptor <- Fox.successful(RemoteSourceDescriptor(path.toUri, None))
+      fullPath <- Fox.successful(path.resolve(layerDirectory))
+      remoteSourceDescriptor <- Fox.successful(RemoteSourceDescriptor(fullPath.toUri, None))
       vaultPath <- dataVaultService.getVaultPath(remoteSourceDescriptor) ?~> "dataVault.setup.failed"
-      layersWithVoxelSizes <- (new NgffExplorer).explore(vaultPath, None)
+      layersWithVoxelSizes <- explorer.explore(vaultPath, None)
       (layers, voxelSize) <- adaptLayersAndVoxelSize(layersWithVoxelSizes, None)
-      zarrLayers = layers.map(_.asInstanceOf[ZarrDataLayer])
-      relativeLayers = makePathsRelative(zarrLayers).toList
+      relativeLayers = makeLayersRelative(layers)
       dataSource = new DataSource(dataSourceId, relativeLayers, voxelSize)
     } yield dataSource
 
-  private def makePathsRelative(layers: Seq[ZarrDataLayer]): Seq[DataLayer] =
-    layers.map(l => l.copy(mags = l.mags.map(m => m.copy(path = m.path.map(_.split("/").takeRight(2).mkString("/"))))))
+  private def makeZarrDataLayerPathRelative(layer: ZarrDataLayer): DataLayer =
+    layer.copy(mags = layer.mags.map(m => m.copy(path = m.path.map(_.split("/").takeRight(2).mkString("/")))))
+
+  private def makeZarrSegmentationLayerPathRelative(layer: ZarrSegmentationLayer): DataLayer =
+    layer.copy(mags = layer.mags.map(m => m.copy(path = m.path.map(_.split("/").takeRight(2).mkString("/")))))
```

Comment:
what is this doing? :thinking: 

Also, should probably be extracted to makeMagPathRelative since this is used twice here.

For N5/precomputed this is not needed?

Reply:
Is used for zarr because of the different directory format (for ngff and zarr, there is an additional directory e.g. color)